Mathlib/Tactic/Positivity/Core.lean

```diff
@@ -252,28 +251,48 @@ where `a` is a numeral. -/
 def compareHyp (e : Q($α)) (ldecl : LocalDecl) : MetaM (Strictness zα pα e) := do
   have e' : Q(Prop) := ldecl.type
   match e' with
-  | ~q(@LE.le _ $_a $lo $hi) =>
-    guard <| ← isDefEq e hi
-    compareHypLE zα pα lo e (.fvar ldecl.fvarId)
-  | ~q(@LT.lt _ $_a $lo $hi) =>
-    guard <| ← isDefEq e hi
-    compareHypLT zα pα lo e (.fvar ldecl.fvarId)
-  | ~q(($lo : $α') = $hi) =>
-    let .true ← isDefEq α α' | return .none
-    let p : Q($lo = $hi) := .fvar ldecl.fvarId
-    match ← compareHypEq zα pα e lo hi p with
-    | .none => compareHypEq zα pα e hi lo (q(Eq.symm $p) : Expr)
-    | result => pure result
-  | ~q(($a : $α') ≠ $b) =>
-    let .true ← isDefEq α α' | return .none
-    if ← isDefEq q((0 : $α)) a then
-      let .true ← isDefEq e b | return .none
-      let p : Q(0 ≠ $e) := .fvar ldecl.fvarId
-      pure (.nonzero q(Ne.symm $p))
-    else
-      let .true ← isDefEq q((0 : $α)) b | return .none
-      let .true ← isDefEq e a | return .none
-      pure (.nonzero (.fvar ldecl.fvarId))
+  | ~q(@LE.le.{u} $β $_le $lo $hi) =>
+    let .defEq (_ : $α =Q $β) ← isDefEqQ α β | return .none
```

Comment:
Another way around the bug here is to use
```suggestion
    let .defEq _← isDefEqQ (u := u) α β | return .none
```
up to you which you prefer

Reply:
I think your version is better after all, since my initial guess of the right universe was wrong!

Reply:
It's annoying having to deal with universes so explicitly, and also having to deal with all the implicit parameters... oh well